test/unit/node/generated-code-dependencies.js

```diff
@@ -274,7 +274,56 @@ suite('context test', function() {
       done();
     });
   });
-
+*/
+  test('sourceMappingURL from --source-maps=true and deep directories', function(done){
+    var deepDirectory = 'test/wiki/CompilingOffline/deepDirectory/';
+    var pwd = process.cwd();
+    process.chdir(deepDirectory);
+    var executable = '../../../../traceur';
+    var inputFilename = './src/js/app.js';
+    var cmd = executable + ' --source-maps ' +
+        '--source-root=false --out ./dist/js/bundle.js ' + inputFilename;
+        console.log('execing  ' + cmd)
+    var child = exec(cmd, function(error, stdout, stderr) {
+      //assert.isNull(error);
+      console.log('and were back')
+      process.chdir(pwd);
+      logOnError(cmd, true, stdout, stderr);
+      console.log('ok')
+      var tempFileName = resolve(deepDirectory + '/dist/js/bundle.js');
+      console.log('reading ' + tempFileName + ' from  ' + process.cwd());
+      var transcoded = fs.readFileSync(tempFileName, 'utf-8');
+      var m = /\/\/#\s*sourceMappingURL=(.*)/.exec(transcoded);
+      assert(m, 'sourceMappingURL appears in the output');
+      var sourceMappingURL = m[1];
+      console.log('test found sourceMappingURL ' + sourceMappingURL);
+      var basepath = path.dirname(tempFileName);
+      var expected = 'bundle.js.map';
+      assert.equal(sourceMappingURL, expected);
+      tempMapName = tempFileName + '.map';
+      var map = JSON.parse(fs.readFileSync(tempMapName, 'utf-8'));
+      console.log('map', map)
+      var actualSourceRoot = map.sourceRoot;
+      // Trailing slash as in the example,
+      // https://github.com/mozilla/source-map
+      assert.equal(undefined, actualSourceRoot, 'has undefined sourceroot');
+      var absoluteInputFilename = path.resolve(deepDirectory, inputFilename);
+      var inputRelativeToOutput =
+          path.relative(path.dirname(tempFileName), absoluteInputFilename);
+      console.log('absoluteInputFilename:' + absoluteInputFilename);
+      var foundInput = map.sources.some(function(name) {
+        if (inputRelativeToOutput === name) {
+          // The 'sources' entry is relative
+          assert.equal(name.indexOf('.'), 0);
+          return true;
+        }
+      });
+      assert(foundInput, 'the inputFilename '+ inputRelativeToOutput +
+          ' is one of the sourcemap sources ' + map.sources.join(','));
+      done();
+    });
+  });
+/*
```

Comment:
I failed to commit my changes, then I pushed. So my debug WIP got committed rather than my PR, sorry.